rosbag2_storage/include/rosbag2_storage/storage_traits.hpp

```diff
@@ -17,12 +17,16 @@
 #define ROSBAG2_STORAGE__STORAGE_TRAITS_HPP_
 
 #include "rosbag2_storage/storage_interfaces/base_io_interface.hpp"
-#include "rosbag2_storage/storage_interfaces/read_only_interface.hpp"
-#include "rosbag2_storage/storage_interfaces/read_write_interface.hpp"
 
 namespace rosbag2_storage
 {
 
+namespace storage_interfaces
+{
+class ReadWriteInterface;
```

Comment:
Why do you need these forward declarations now? Is this leftover from a previous iteration / should this file be reverted?

Reply:
These forward declarations are still needed since the templates below reference `ReadWriteInterface` and `ReadOnlyInterface` and that's the only way to avoid a circular dependency since `ReadWriteInterface` and `ReadOnlyInterface` include `storage_traits.hpp`.